chalk-rust-ir/src/lib.rs

```diff
@@ -103,6 +103,23 @@ pub struct StructFlags {
     pub fundamental: bool,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct FnDefDatum<I: Interner> {
+    pub id: FnDefId<I>,
+    pub binders: Binders<FnDefDatumBound<I>>,
+    pub flags: FnDefFlags,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Fold, HasInterner)]
+pub struct FnDefDatumBound<I: Interner> {
```

Comment:
I feel like we are going to need .. well, eventually we'll have to add ABI and other details, but maybe we can add that together with adding the same details to the function pointer types